siac/walletcmd.go

```diff
@@ -373,6 +381,61 @@ Siafund Claims:      %v SC
 `, encStatus, lockStatus, sc, usc, status.ConfirmedSiacoinBalance, status.SiafundBalance, status.SiacoinClaimBalance)
 }
 
+// wallettransactionscmd lists all of the transactions related to the wallet,
+// providing a net flow of siacoins and siafunds for each.
+func wallettransactionscmd() {
+	wtg := new(api.WalletTransactionsGET)
+	err := getAPI("/wallet/transactions?startheight=0&endheight=10000000", wtg)
+	if err != nil {
+		fmt.Println("Could not fetch transaction history:", err)
+		return
+	}
+
+	fmt.Println("[height]\t[transaction id]\t\t\t\t\t\t\t[net siacoin transfer]\t[net siafund transfer]")
+	txns := append(wtg.ConfirmedTransactions, wtg.UnconfirmedTransactions...)
+	for _, txn := range txns {
+		// Determine the number of outgoing siacoins and siafunds.
+		var outgoingSiacoins types.Currency
+		var outgoingSiafunds types.Currency
+		for _, input := range txn.Inputs {
+			if input.FundType == types.SpecifierSiacoinInput && input.WalletAddress {
+				outgoingSiacoins = outgoingSiacoins.Add(input.Value)
+			}
+			if input.FundType == types.SpecifierSiafundInput && input.WalletAddress {
+				outgoingSiafunds = outgoingSiafunds.Add(input.Value)
+			}
+		}
+
+		// Determine the number of incoming siacoins and siafunds.
+		var incomingSiacoins types.Currency
+		var incomingSiafunds types.Currency
+		for _, output := range txn.Outputs {
+			if output.FundType == types.SpecifierMinerPayout {
+				incomingSiacoins = incomingSiacoins.Add(output.Value)
+			}
+			if output.FundType == types.SpecifierSiacoinOutput && output.WalletAddress {
+				incomingSiacoins = incomingSiacoins.Add(output.Value)
+			}
+			if output.FundType == types.SpecifierSiafundOutput && output.WalletAddress {
+				incomingSiafunds = incomingSiafunds.Add(output.Value)
+			}
+		}
+
+		// Convert the siacoins to a float.
+		incomingSiacoinsFloat, _ := new(big.Rat).SetFrac(incomingSiacoins.Big(), types.SiacoinPrecision.Big()).Float64()
+		outgoingSiacoinsFloat, _ := new(big.Rat).SetFrac(outgoingSiacoins.Big(), types.SiacoinPrecision.Big()).Float64()
+
+		// Print the results.
+		fmt.Printf("%v\t\t%v\t %.2f SC\t\t", txn.ConfirmationHeight, txn.TransactionID, incomingSiacoinsFloat-outgoingSiacoinsFloat)
```

Comment:
see if you can use a width specifier to do the alignment, something like `"%10v %v %10.2f SC"`